### ISSUE TITLE
salt.states.git does not set permissions correctly

### ISSUE THREAD

**Issue by loz-hurst** [saltstack/salt#57452]
git clone uses the umask to control the initial permissions on cloned files.  When cloning sensitive files from our private git servers I want to clone them with a secure permissions (which I can do on the command line by setting umask (0077 or 0007)).  Git also provides a mechanism for overriding the effective umask via the `core.sharedRepository` configuration option (https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-config.html).

Neither `git.latest` or `git.cloned` provide the option for specifying the git command's environmental umask or any configuration options (e.g. the command line to give user read and write and deny group/other on clone would be `git clone --config core.sharedRepository=0600`).  The configuration version can be set on the cloned repository with git.config_set but only after a clone has been done requiring another state to fix the incorrect permissions, leaving a window when the insecure files are world readable and taking 3 states to do what could be done directly with the right `git clone` command.

**Comment by loz-hurst** [comment]
I had a few searches through the open issues and could not find one for this already, apologies if it does exist and I missed it.

**Comment by Akm0d** [comment]
Could the git cloning state be used in conjunction with [salt.states.file](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.file.html#salt.states.file.directory) to manage permissions on the cloned directory?

**Comment by oeuftete** [comment]
@loz-hurst The git module in places supports using `git_opts` (with [`clone`](https://docs.saltstack.com/en/master/ref/modules/all/salt.modules.git.html#salt.modules.git.clone), for example) to pass configure options.  If the analogous _states_ were updated to allow a `git_opts` argument as well, that would allow you to handle your use case in one secure step.

Is that the kind of thing you are looking for here?

**Comment by loz-hurst** [comment]
@Akm0d I tried that, it cannot be done securely - there is a window between the clone and the file state being applied when the files are world-readable on disk, which is not acceptable in my use-case.  The best work-around I have is to cmd.script it with a script that sets the umask and then runs `git clone`.

**Comment by loz-hurst** [comment]
@oeuftete Yes, that would be an ideal solution.  I have a very specific use-case but allowing us to set git_opts in the state would be useful in lots of situations I expect.  I'm using `git.latest` with this repo, so for me `git.latest` and `git.clone` would need to support the option.  I see that `git.pull` in the module supports the option too, so I presume this is fairly straight-forward to implement although I've not looked at the code yet (little busy at the moment...).
